### PR TITLE
Preliminary handling for missing content in editing lessons and quizzes

### LINKED ISSUE #7327: ensure that quizzes and lessons are not modified when they reference missing content
### Observed behavior

current behavior might be that quizzes and lessons might actually be modified, deleting the missing resource reference, if those lessons or quizzes are edited on a device without all the content

### Expected behavior

let's double-check the behavior and ensure that no data modification occurs

### User-facing consequences

data loss

### Steps to reproduce

* create a lesson or quiz on one device
* sync to another device which lacks the content
* edit lesson and quiz
* sync back and ensure that no references were deleted

### Context

0.14.0

### PR DESCRIPTION
**Pull request by rtibbles**
### Summary
* Stops frontend from auto-removing missing content
* Removes separate ancestor fetching to avoid errors on content management page for lessons
* Adds missing resource placeholder for lesson resource details
* Allows saving resources that are missing
* Adds a message in quiz preview when trying to render a question that is missing
* Adds an icon and tooltip to each question that is missing its exercise

### Reviewer guidance
Quiz preview
Screenshot from 2020-08-03 17-51-05

Lesson resources
Screenshot from 2020-08-03 17-52-18

### References
Fixes #7327

…

----

### Contributor Checklist

PR process:

- [x] PR has the correct target branch and milestone
- [x] PR has 'needs review' or 'work-in-progress' label
- [x] If PR is ready for review, a reviewer has been added. (Don't use 'Assignees')
- [ ] If this is an important user-facing change, PR or related issue has a 'changelog' label
- [ ] If this includes an internal dependency change, a link to the diff is provided

Testing:

- [x] Contributor has fully tested the PR manually
- [x] If there are any front-end changes, before/after screenshots are included
- [ ] Critical user journeys are covered by Gherkin stories
- [ ] Critical and brittle code paths are covered by unit tests

### Reviewer Checklist

- Automated test coverage is satisfactory
- PR is fully functional
- PR has been tested for [accessibility regressions](http://kolibri-dev.readthedocs.io/en/develop/manual_testing.html#accessibility-a11y-testing)
- External dependency files were updated if necessary (`yarn` and `pip`)
- Documentation is updated
- Contributor is in AUTHORS.md

### CODE REVIEW

**Comment by indirectlylit** [comment]
partially addresses https://github.com/learningequality/kolibri/issues/7325

File: kolibri/plugins/coach/assets/src/views/plan/LessonSummaryPage/ResourceListTable.vue [hunk 1]
```diff
@@ -28,32 +28,42 @@
...
+                <CoachContentLabel
+                  class="coach-content-label"
+                  :value="getCachedResource(resource.contentnode_id).num_coach_contents"
+                  :isTopic="false"
+                />
+              </template>
+              <template v-else>
+                <p>
+                  <KIcon icon="warning" :style=" { fill: $themePalette.orange.v_400 }" />
+                  {{ resourceMissingText }}
```
Comment (indirectlylit) [inline kolibri/plugins/coach/assets/src/views/plan/LessonSummaryPage/ResourceListTable.vue:58]: can these be implemented with `KLabeledIcon`?

Comment (nucleogenesis) [reply kolibri/plugins/coach/assets/src/views/plan/LessonSummaryPage/ResourceListTable.vue:58]: Also there is a `color` prop on `KIcon`

Comment (rtibbles) [reply kolibri/plugins/coach/assets/src/views/plan/LessonSummaryPage/ResourceListTable.vue:58]: Using KLabeledIcon makes the appearance and formatting significantly worse:

Screenshot from 2020-08-04 11-48-37

vs with just a `<p>`:
Screenshot from 2020-08-04 11-49-24

**Review by micahscopes — APPROVED** [review]
I tested this pretty thoroughly and it works very well as far as I can tell!  There was one edge case, but Richard and I were able to debug it over a call and he figured out how to fix it!

There was just one minor issue with the display of the missing content tooltip in the "coach --> plan --> edit quiz" view, but this is probably a separate issue, as I can't see that `KTooltip` is used any differently in this PR than it is elsewhere in Kolibri: 
fluttering tooltip

File: kolibri/plugins/coach/assets/src/views/plan/LessonSummaryPage/ResourceListTable.vue (hunk 1 above)
Comment (nucleogenesis) [reply kolibri/plugins/coach/assets/src/views/plan/LessonSummaryPage/ResourceListTable.vue:58]: Is the bad one a KLabeledIcon in a `<p>` or by itself?

Comment (rtibbles) [reply kolibri/plugins/coach/assets/src/views/plan/LessonSummaryPage/ResourceListTable.vue:58]: The first one is with the KLabeledIcon and I think it gives a much worse result than the second one which is using a `<p>` tag

Comment (rtibbles) [reply kolibri/plugins/coach/assets/src/views/plan/LessonSummaryPage/ResourceListTable.vue:58]: Oh, the KLabeledIcon is by itself, sorry, no p

Comment (indirectlylit) [reply kolibri/plugins/coach/assets/src/views/plan/LessonSummaryPage/ResourceListTable.vue:58]: Fine to leave for now, but FYI that looks like something was misconfigured. From `KLabeledIcon`:

```scss
  .debug-warning > svg {
    // if you see this, you need to pass the icon into the slot
    border: 1px solid red;
  }
```

**Pull request merged**